### PR TITLE
Http automatons fixes

### PR DESCRIPTION
**Pull request by gpotter2**
Bunch of edits I for some reason never published
- close http automatons properly
- include the iptables code directly in the request handler

### CODE REVIEW

File: scapy/layers/http.py [hunk 1]
```diff
@@ -616,12 +619,18 @@ def http_request(host, path="/", port=80, timeout=3,
     }
     http_headers.update(headers)
     req = HTTP() / HTTPRequest(**http_headers)
-    tcp_client = TCP_client.tcplink(HTTP, host, 80)
+    tcp_client = TCP_client.tcplink(HTTP, host, 80, debug=verbose)
     ans = None
+    if iptables:
+        ip = tcp_client.atmt.dst
+        iptables_rule = "iptables -%c INPUT -s %s -p tcp --sport 80 -j DROP"
```
Comment (guedou) [inline scapy/layers/http.py]: Could you add `--tcp-flags ALL RST` ?

Comment (guedou) [reply scapy/layers/http.py]: Also, I think that this should be `OUTPUT` instead of ` INPUT`

Comment (gpotter2) [reply scapy/layers/http.py]: I copy pasted the rule you made in the HTTP tests.
When I try using output *or* tcp flags, it fails..

Comment (guedou) [reply scapy/layers/http.py]: What is the goal of this rule? Drop everything or only RST segments?

**Pull request merged**